src/citrine/waiting/waiting.py

```diff
@@ -0,0 +1,77 @@
+from time import time, sleep
+from typing import Callable
+from pprint import pprint
+from citrine._rest.collection import Collection
+from citrine.resources.workflow_executions import (
+    WorkflowExecution,
+    WorkflowExecutionStatus,
+)
+
+
+def _wait_until(condition: Callable[[], bool], timeout: float = 900.0, interval: float = 3.0):
+    """Poll until the provided condition is truthy or the timeout is reached."""
+    start = time()
+    while not condition() and time() - start < timeout:
+        sleep(interval)
+    assert condition(), "Timeout reached, but condition is still False."
+
+
+def _print_validation_status(
+    status: str, start_time: float, line_start: str = "", line_end: str = "\r"
+):
+    print(
+        f"{line_start}Status = {status:<25}Elapsed time"
+        " = {str(int(time() - start_time)).rjust(3)}s",
+        end=line_end,
+    )
+
+
+def _pretty_execution_status(status: WorkflowExecutionStatus):
+    status_text = status.status
+    output_text = status_text if status_text != "InProgress" else "In progress"
+    return output_text
+
+
+def _print_execution_status(
+    status: WorkflowExecutionStatus, start_time: float, line_end: str = "\r"
+):
+    print(
+        f"Status = {_pretty_execution_status(status):<25}Elapsed time"
+        " = {str(int(time() - start_time)).rjust(3)}s",
+        end=line_end,
+    )
+
+
+def wait_until_validated(
```

Comment:
This method name implies to me that the module should always be VALID after it completes successfully.  Maybe something like `wait_while_validating` would imply less about the final state.